rfd/0073-public-image-registry.md

```diff
@@ -0,0 +1,249 @@
+
+### Migration and Deprecation
+* Using the list of public teleport images defined [below](#appendix-b-teleport-public-images), the terraform infrastructure needed for these registries will be created according to the standards defined in[artifact storage standards](https://github.com/gravitational/cloud/blob/9124947fdfb0773fa9bd567160481bed4ec84b7e/rfd/0017-artifact-storage-standards.md#levels).
+* Existing CI/CD pipelines and tooling will be updated to push to both Quay.io and ECR. 
+* Documentation will be updated to reflect the new registry location. 
+* Quay.io images will be deprecated and removed in reverse level order. (Bronze -> Silver -> Gold)
+
+Gold standard images (teleport, teleport-ent, etc...) will continue to exist and be pushed to Quay.io for the foreseeable future. [Artifact Mirroring](https://github.com/gravitational/teleport/commit/2262efbb25463ccc135553d43293f6d8aee22ba2) will go into more details. 
+
+### Alternatives
+#### Self hosted with Docker Distribution or Harbor
+The [Docker Distribution](https://github.com/distribution/distribution) registry is an open source implementation of the [OCI Distribution](https://github.com/opencontainers/distribution-spec) specification. [Harbor](https://goharbor.io/) is an OSS, _all-in-one_, registry solution that is built on top of the docker registry. Harbor has a built-in UI, support for OIDC authenticatio and many more [additional features](https://goharbor.io/docs/2.5.0/). 
+
+While Harbor provides a maximally featured container registry solution, it also incurs an increased operational overhead that Teleport didn't have with Quay.io.
+
+#### Custom Registry w/ CloudFront Functions
+A minimal, _oci-compatible_ registry could be implemented through just CloudFront functions. This registry would only support reading. This would reduce the operational complexity of the current strategy to AWS specific components. Additional components would be needed to be developed in order to push the image to the S3 bucket but could be implemented as just another step in the existing CI/CD pipeline.
+
+A negative to this solution is the lack of features that come standard with other registry solutions. This includes, but is not limited to, vulnerability scanning and tag immutability. Additionally, discoverability would be a missing feature from this solution.
+
+### Appendix A: Example Terraform
+```hcl
+# Internal Repository for Teleport
+resource "aws_ecr_repository" "teleport" {
+  name                 = "gravitational/teleport"
+  image_tag_mutability = "IMMUTABLE"
+
+  encryption_configuration {
+    encryption_type = "KMS"
+    kms_key         = aws_kms_key.ecr_key.arn
+  }
+
+  image_scanning_configuration {
+    scan_on_push = true
+  }
+}
+
+# Public Repository for Teleport
+resource "aws_ecrpublic_repository" "teleport" {
+  repository_name = "teleport"
```

Comment:
`teleport` or `gravitational`?  The diagram above says `gravitational`

We should probably squat both, even if we only use one.

Reply:
The public repository would already be using `gravitational` as it's registry alias, therefore `teleport` in this case refers to public.ecr.aws/gravitational/*teleport*. 

I'm open to having `gravitational`, `teleport` or both as the registry alias for public ECR. I would note the recent terraform provider went with `gravitational/teleport`.

Reply:
Whoops -- my bad.  I was mistaking the repository for the org (or "registry alias"?).  Where is the org/registry alias defined?  I don't see that in this example TF.

Reply:
I don't have permissions currently. I will start on the `core-release-engineer-admin` role since regardless of what we end up doing that will be useful.

Reply:
Commenting here because @logand22 reached out to me asking if I had opinions on rebranding images from `public.ecr.aws/gravitational/teleport` to `public.ecr.aws/teleport/teleport`, thereby dropping the `gravitational` brand from container images. Personally, I find that a complete rebrand across the board is not at the top of the priority list so imo, going with `/gravitational/teleport` is preferable for consistency (mainly less confusing for users). I agree that we should squat `teleport` too if/when we decide to completely swap everything over.